src/components/core/TextArea/TextArea.tsx

```diff
@@ -10,16 +10,14 @@ type TextAreaProps = {
   placeholder: string;
   defaultValue?: string;
   autoFocus?: boolean;
+  textAreaHeight?: string;
 };
 
-export function TextArea({
-  className,
-  onChange = noop,
-  placeholder,
-  defaultValue,
-  autoFocus = false,
-}: TextAreaProps) {
-  return (
+export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
```

Comment:
nice use of forwardRef! 🔥 

so we still have a vanilla `TextAreaWithCharCount` component that uses this `TextArea` component. this doesn't pass in a ref – guessing that's ok?

![image](https://user-images.githubusercontent.com/12162433/159343534-d0c2ef74-2b47-4714-8e77-b1ac62a66d02.png)

Reply:
It works (e.g. doesn't break anything) but I did come across [this rule](https://deepscan.io/docs/rules/react-useless-forward-ref) so likely not best practice.

It seems like this is our [best approach](https://stackoverflow.com/a/69127914):

```jsx
export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
  (
    { className, onChange = noop, placeholder, defaultValue, autoFocus = false, textAreaHeight },
    ref
  ) => {
    const _ref = useRef<HTMLTextAreaElement>(null);
    const textAreaRef = ref || _ref;
...
```

Alternatively, we could make a separate component that doesn't use forwardRef but is otherwise the same... Although the above feels better to me.

Reply:
+1 your suggestion feels ideal. thanks for digging!